website/docs/d/resource_list.html.markdown

```diff
@@ -0,0 +1,299 @@
+    "dns1" = "8.8.8.8"
+    "dns2" = "8.8.4.4"
+    "dns_suffix" = ""
+    "edge_gateway" = "gw-datacloud"
+    "gateway" = "192.168.2.1"
+    "href" = "https://vcd.example.com/api/network/04915abf-0c91-4919-878e-0f292e032e2b"
+    "id" = "urn:vcloud:network:04915abf-0c91-4919-878e-0f292e032e2b"
+    "name" = "net-datacloud-r"
+    "netmask" = "255.255.255.0"
+    "org" = "datacloud"
+    "shared" = false
+    "static_ip_pool" = [
+      {
+        "end_address" = "192.168.2.100"
+        "start_address" = "192.168.2.2"
+      },
+    ]
+    "vdc" = "vdc-datacloud"
+  }
+  {
+    "description" = ""
+    "dhcp_pool" = []
+    "dns1" = ""
+    "dns2" = ""
+    "dns_suffix" = ""
+    "edge_gateway" = "gw-datacloud"
+    "gateway" = "192.168.3.1"
+    "href" = "https://vcd.example.com/api/network/2cc713b1-134f-4f21-9208-79f1e4f3ee36"
+    "id" = "urn:vcloud:network:2cc713b1-134f-4f21-9208-79f1e4f3ee36"
+    "name" = "net-datacloud-r2"
+    "netmask" = "255.255.255.0"
+    "org" = "datacloud"
+    "shared" = false
+    "static_ip_pool" = [
+      {
+        "end_address" = "192.168.3.50"
+        "start_address" = "192.168.3.2"
+      },
+    ]
+    "vdc" = "vdc-datacloud"
+  }
+]
+*/
+```
+
+## Example 6 - List of resources
+
+```hcl
+data "vcd_resource_list" "list_of_resources" {
+  name          = "list_of_resources"
+  resource_type = "resources"
+}
+
+# Shows the list of resource types for VCD provider
+output "resource_list" {
+  value = data.vcd_resource_list.list_of_resources.list
+}
+```
+
+
+## Example 7 - List of catalog items
+
+```hcl
+data "vcd_resource_list" "list_catalog_items" {
+  name          = "list_of_catalog_items"
+  resource_type = "vcd_catalog_item" 
+  parent        = "cat-datacloud"      # name of the catalog to be listed
+  list_mode     = "name"
+}
+
+# Shows the list of all catalog items
+output "catalog_items" {
+  value = data.vcd_resource_list.list_of_catalog_items.list
+}
+
+/*
+output: 
+catalog_items = [
+  "photon-hw11",
+  "Linux1",
+  "debian-7",
+]
+*/
+```
+
+## Example 8 - List of LB virtual servers
+
+```hcl
+data "vcd_resource_list" "list_lb_virtual_servers" {
+  name          = "list_of_virtual_servers"
+  resource_type = "vcd_lb_virtual_server" 
+  parent        = "gw-datacloud"      # name of the edge gateway to be listed
+  list_mode     = "name"
+}
+
+# Shows the list of all LB virtual servers
+output "lb_virtual_servers" {
+  value = data.vcd_resource_list.list_lb_virtual_servers.list
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional) The name of organization to use, optional if defined at provider level
+* `vdc` - (Optional) The name of VDC to use, optional if defined at provider level
+* `name` - (Required) An unique name to identify the data source
+* `resource_type` (Required) Which resource we want to list. Supported keywords are:
+    * `resources`  (list the resource types in the provider)
+    * `vcd_org`
+    * `vcd_external_network`
```

Comment:
We should add the NSX-T resource support from 3.0.0 too.
CC: @Didainius 

BTW, in general I'm concerned that this resource is manually programmed and adds a custom programming effort with each new added resource in the future. Is there any way to program it in such a way that existing resources are automatically scanned and exposed, instead?

Reply:
We went through this argument in PR #415, and that was the main reason for it being abandoned. 
Listing capability should be a trait of a new resource just like import.
The corresponding function for each entity should be implemented in govcd and added in this data source.

The goal of this resumed feature is to facilitate migration.

Reply:
> We should add the NSX-T resource support from 3.0.0 too.

The NSXT VDC and external network v2 are already found with the corresponding old resources. 
I am looking at the possibility of differentiating these resources with separate lists. The fact us that even in the UI they are collected together

Reply:
Documentation updated